web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java

```diff
@@ -46,23 +47,25 @@
 	 */
 	public IpAddressMatcher(String ipAddress) {
 
-		if (ipAddress.indexOf('/') > 0) {
-			String[] addressAndMask = StringUtils.split(ipAddress, "/");
-			ipAddress = addressAndMask[0];
-			nMaskBits = Integer.parseInt(addressAndMask[1]);
-		}
-		else {
-			nMaskBits = -1;
-		}
-		requiredAddress = parseAddress(ipAddress);
+		String [] addressAndMask = preprocessAddressAndMask(ipAddress);
+		String ipAddr = addressAndMask[0];
+		nMaskBits = Integer.parseInt(addressAndMask[1]);
+		requiredAddress = parseAddress(ipAddr);
+		Assert.isTrue(requiredAddress.getAddress().length * 8 >= nMaskBits,
+				String.format("Mask bits too long for this address %s", ipAddress));
 	}
 
 	public boolean matches(HttpServletRequest request) {
 		return matches(request.getRemoteAddr());
 	}
 
 	public boolean matches(String address) {
-		InetAddress remoteAddress = parseAddress(address);
+		String [] addressAndMask = preprocessAddressAndMask(address);
```

Comment:
The address passed into this method doesn't have a mask, so the preprocessing is unnecessary here.

As such, the `preprocessAddressAndMask` method is probably also unnecessary for now.

Reply:
So does this mean that we don't have to check `remote address` anymore, since there's no mask that will be assigned there?

Reply:
Because of this line:

```java
if ((remAddr[i] & mask[i]) != (reqAddr[i] & mask[i])) {
```

The code needs to ensure that both `remAddr` and `reqAddr` are not too short. It makes sense to me that the code would verify the required address in the constructor and the remote address in the `matches` method.

Reply:
I assume you're referring to this [line](https://github.com/spring-projects/spring-security/blob/3ea9d376b2e6e0d46828de779fc5eb00ab0a8699/web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java#L90). I can add a comparison that the `remAddr.length * 8 >= nMaskBits`, but I'm having trouble coming up w/ a too short IP address. Stuff like `192.168.1` and shorter variants seem to be matching just fine.

Reply:
Oops, I was behind a version. I'm referring to [this line](https://github.com/spring-projects/spring-security/blob/3ea9d376b2e6e0d46828de779fc5eb00ab0a8699/web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java#L84).

I wonder if it would break if you configured to match against an IPv6 address and called `matches` with an IPv4 address?

Reply:
@jzheaux In that case [this line](https://github.com/spring-projects/spring-security/blob/3ea9d376b2e6e0d46828de779fc5eb00ab0a8699/web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java#L67) will realize that the classes of the IP addresses are different (`java.net.Inet6Address` vs `java.net.Inet4Address`) and then simply return `False`. IMO this feels more correct than an `IllegalArgumentException` stating that a mask length is too long, which isn't indicative that the addresses you're comparing are of different types. Unless it is valid to try to mix and match ipv4 and ipv6 addresses?

Reply:
@jzheaux In that case [this line](https://github.com/spring-projects/spring-security/blob/3ea9d376b2e6e0d46828de779fc5eb00ab0a8699/web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java#L67) will realize that the classes of the IP addresses are different (`java.net.Inet6Address` vs `java.net.Inet4Address`) and then simply return `False`. IMO this feels more correct than an `IllegalArgumentException` stating that a mask length is too long, which isn't indicative that the addresses you're comparing are of different types. Unless it is valid to try to mix and match ipv4 and ipv6 addresses?

Reply:
No, it's not a valid use case -- was just thinking out loud about how you might make the remote address byte array too long.

I agree that an `IPv4Address` instance will always be 4 bytes long and an `IPv6Address` will always be 16 bytes long, so there's no need to check the length of remote address.

Reply:
Thanks @jzheaux, any other feedback?